### PR TITLE
Removes unnecessary references to AppEngineFlexibleDeploymentConfiguration

### PR DESCRIPTION
**Pull request by joaoandremartins**
I don't remember exactly why I went for this design, but I can clearly see now it wasn't a great decision.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 1]
```diff
@@ -215,7 +211,7 @@ private boolean isValidConfigurationFile(String path) {
   /**
    * Validates the configuration and turns on/off any necessary warnings.
    */
-  private void validateConfiguration() {
+  private void toggleWarnings() {
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: displayWarnings or showWarnings?  

'toggle' means to switch state, which I guess we don't always do here.

**Comment by patflynn** [comment]
@joaoandremartins cool, but maybe include a bit more context as to why it was a bad decision in the PR description. :)

**Comment by joaoandremartins** [comment]
Sorry about the lack of context.
I was passing a deployment configuration to the facet editor, which doesn't make a lot of sense.
I suspect I did this to enable the "Add Framework Support -> Runnable deployment run configuration out of the box" flow, but I don't think this even achieved that.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 2]
```diff
@@ -78,13 +77,13 @@
   private JLabel noDockerfileLabel;
   private JLabel errorIcon;
   private JLabel errorMessage;
-  private AppEngineDeploymentConfiguration deploymentConfiguration;
+  private AppEngineFlexibleFacetConfiguration facetConfiguration;
   private AppEngineHelper appEngineHelper;
 
-  FlexibleFacetEditor(@Nullable AppEngineDeploymentConfiguration deploymentConfiguration,
+  FlexibleFacetEditor(AppEngineFlexibleFacetConfiguration facetConfiguration,
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: facetConfiguration @NotNull still?

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: Oh I'm never sure which one is (not) required.
Aren't fields non-null by default? Or is it the other way around?

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: I don't think either is default. Having these just aids in static analysis. I.e. the IDE will give useful warnings if you pass null to an @NotNull argument.

**Comment by joaoandremartins** [comment]
@etanshaul @patflynn ptal

**Pull request merged**